File: AngularUI/src/app/core/services/auth.service.ts
import { TokenService } from './token.service';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { apiUrl } from 'src/api';
import { SingleResponseModel } from '../models/response/singleResponseModel';
import { LoginModel } from '../models/user/loginModel';
import { RegisterModel } from '../models/user/registerModel';
import { TokenModel } from '../models/user/tokenModel';
import { LocalStorageService } from './local-storage.service';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  constructor(
    private httpClient: HttpClient,
    private localStorageService: LocalStorageService,
    private tokenService: TokenService
  ) {}

  login(user: LoginModel): Observable<SingleResponseModel<TokenModel>> {
    let newPath = apiUrl + 'auth/login';

    return this.httpClient.post<SingleResponseModel<TokenModel>>(newPath, user);
  }

  register(user: RegisterModel): Observable<SingleResponseModel<TokenModel>> {
    let newPath = apiUrl + 'auth/register';

    return this.httpClient.post<SingleResponseModel<TokenModel>>(newPath, user);
  }

  logout() {
    if (this.isAuthenticated()) {
      this.tokenService.removeToken();
      return true;
    }

    return false;
  }

  isAuthenticated(): boolean {
    if (this.localStorageService.controlItem('token')) {
      return true;
    }

    return false;
  }

}

File: flutter_ui/lib/core/services/authService.dart
import 'dart:convert';
import 'package:flutter_ui/core/interceptors/authInterceptor.dart';
import 'package:flutter_ui/core/models/response/singleResponseModel.dart';
import 'package:flutter_ui/core/models/user/loginModel.dart';
import 'package:flutter_ui/core/models/user/registerModel.dart';
import 'package:flutter_ui/core/models/user/tokenModel.dart';
import 'package:flutter_ui/core/services/service.dart';
import 'package:flutter_ui/core/services/sessionService.dart';
import 'package:flutter_ui/environments/api.dart';

class AuthService extends Service {
  SessionService _sessionService = SessionService();

  Future<SingleResponseModel<TokenModel>> login(LoginModel user) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "auth/login"),
      body: user.toJson(),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<TokenModel>.fromJson(jsonData);
  }

  Future<SingleResponseModel<TokenModel>> register(RegisterModel user) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "auth/register"),
      body: user.toJson(),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<TokenModel>.fromJson(jsonData);
  }

  bool logout() {
    if (isAuthenticated() == true) {
      _sessionService.remove("token");

      return true;
    }

    return false;
  }

  bool isAuthenticated() {
    bool token;

    _sessionService
        .get("token")
        .then((value) => value != null ? token = true : token = false);

    if (token) {
      return true;
    }

    return false;
  }
}
